File: e-kompen-api/app/Models/Absensi.php
<?php
namespace App\Models;
use Illuminate\Database\Eloquent\Model;

class Absensi extends Model {
    protected $table      = 'absensi';
    protected $primaryKey = 'id_absensi';
    public $timestamps    = false;
    protected $fillable   = [
        'id_mahasiswa', 'id_mata_kuliah', 'id_dosen', 'id_admin',
        'tanggal', 'status', 'jml_jam'
    ];

    public function mahasiswa() {
        return $this->belongsTo(Mahasiswa::class, 'id_mahasiswa', 'id_mahasiswa');
    }

    public function mataKuliah() {
        return $this->belongsTo(MataKuliah::class, 'id_mata_kuliah', 'id_mata_kuliah');
    }

    public function dosen() {
        return $this->belongsTo(Dosen::class, 'id_dosen', 'id_dosen');
    }

    public function admin() {
        return $this->belongsTo(Admin::class, 'id_admin', 'id_admin');
    }

    public function pengajuanKompen() {
        return $this->hasMany(PengajuanKompen::class, 'id_absensi', 'id_absensi');
    }
}

File: lib/models/absensi.dart
class Absensi {
  final int idAbsensi;
  final int idMahasiswa;
  final int idMataKuliah;
  final int idDosen;
  final int idAdmin;
  final DateTime tanggal;
  final String status; // 'alpha', 'izin', 'sakit'
  final int jmlJam;

  Absensi({
    required this.idAbsensi,
    required this.idMahasiswa,
    required this.idMataKuliah,
    required this.idDosen,
    required this.idAdmin,
    required this.tanggal,
    required this.status,
    required this.jmlJam,
  });

  factory Absensi.fromJson(Map<String, dynamic> json) {
    return Absensi(
      idAbsensi: json['id_absensi'],
      idMahasiswa: json['id_mahasiswa'],
      idMataKuliah: json['id_mata_kuliah'],
      idDosen: json['id_dosen'],
      idAdmin: json['id_admin'],
      tanggal: DateTime.parse(json['tanggal']),
      status: json['status'],
      jmlJam: json['jml_jam'],
      
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id_absensi': idAbsensi,
      'id_mahasiswa': idMahasiswa,
      'id_mata_kuliah': idMataKuliah,
      'id_dosen': idDosen,
      'id_admin': idAdmin,
      'tanggal': tanggal.toIso8601String(),
      'status': status,
      'jml_jam': jmlJam,
      
    };
  }
}
